src/main/kotlin/reminder/application/ReminderSearcher.kt

```diff
@@ -0,0 +1,19 @@
+package reminder.application
+
+import reminder.domain.*
+import java.util.*
+
+class ReminderSearcher(
+    private val repository: ReminderRepository,
+) {
+
+    /***
+     * Searches a Reminder with the given identifier and returns it if found.
+     * @see Reminder
+     * @return An optional of a reminder instance.
+     */
+    fun search(identifier: ReminderIdentifier): Optional<Reminder> {
+        return repository.search(identifier)
+    }
```

Comment:
From the kotlin in action book:

_In Kotlin, if is an expression, not a statement. The difference between a statement
and an expression is that an expression has a value, which can be used as part of
another expression, whereas a statement is always a top-level element in its enclos-
ing block and doesn’t have its own value. In Java, all control structures are state-
ments. In Kotlin, most control structures, except for the loops (for, do, and do/
while) are expressions._

So, we could change the function to:
```kotlin
fun search(identifier: ReminderIdentifier): Optional<Reminder> = repository.search(identifier)
```
And kotlin can infere the type here, so we could replace it with:
```kotlin
fun search(identifier: ReminderIdentifier) = repository.search(identifier)
```